[Dependency Scanning] Always use a locking diagnostic consumer

artemcm opened the pull request:
Since we enabled parallel dependency scanning by-default, each individual scan needs a diagnostic consumer that is safe to use across many threads

artemcm committed:
[Dependency Scanning] Always use a locking diagnostic consumer

Since we enabled parallel dependency scanning by-default, each individual scan needs a diagnostic consumer that is safe to use across many threads. Deprecate the 'Locking' sub-class, making its behavior the default in the base class.

cachemeifyoucan commented:
Is the base class `DependencyScanDiagnosticCollector` still used? Should we should fold it back so there is only one DiagnosticCollector that is locking?

artemcm committed:
[Dependency Scanning][libSwiftScan] Deprecate scanner-global diagnostic collection

All clients are expected to have switched to the per-scan-query diagnostics using 'swiftscan_dependency_graph_get_diagnostics'

artemcm commented:
@swift-ci smoke test

cachemeifyoucan reviewed (approved):
LGTM
